test/src/integration/onChain.test.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright 2018 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import { TestHelper } from "codechain-test-helper/lib/testHelper";
+import CodeChain from "../helper/spawn";
+import { H256 } from "codechain-sdk/lib/core/H256";
+import { H160 } from "codechain-sdk/lib/core/H160";
+import { U256 } from "codechain-sdk/lib/core/U256";
+import { Header } from "codechain-test-helper/lib/cHeader";
+
+describe("Test onChain communication", () => {
+    let nodeA: CodeChain;
+
+    beforeEach(async () => {
+        nodeA = new CodeChain({ logFlag: true });
+        await nodeA.start();
+    });
+
+    afterEach(async () => {
+        await nodeA.clean();
+    });
+
+    test("OnChain parcel propagation test", async () => {
+        const TH = new TestHelper("0.0.0.0", nodeA.port);
```

Comment:
Could you explain the first argument? Is it an address for listening?

Reply:
The IP address and port are for the target node.
A listening IP and port is 0.0.0.0 and 6602.